regatta-java-core/src/main/java/com/jamf/regatta/core/resolver/AbstractNameResolver.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright JAMF Software, LLC
+ */
+
+package com.jamf.regatta.core.resolver;
+
+import com.google.common.base.Preconditions;
+import io.grpc.Attributes;
+import io.grpc.EquivalentAddressGroup;
+import io.grpc.NameResolver;
+import io.grpc.Status;
+import io.grpc.internal.GrpcUtil;
+import io.grpc.internal.SharedResourceHolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Executor;
+
+public abstract class AbstractNameResolver extends NameResolver {
+    public static final int REGATTA_CLIENT_PORT = 8443;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractNameResolver.class);
+
+    private final Object lock;
+    private final String authority;
+    private final URI targetUri;
+
+    private volatile boolean shutdown;
+    private volatile boolean resolving;
+
+    private Executor executor;
+    private Listener listener;
+
+    public AbstractNameResolver(URI targetUri) {
+        this.lock = new Object();
+        this.targetUri = targetUri;
+        this.authority = targetUri.getAuthority() != null ? targetUri.getAuthority() : "";
```

Comment:
minor: since there is already guava, it may be simplified
```suggestion
        this.authority = MoreObjects.fisrtNonNull(targetUri.getAuthority(), "");
```